src/azul/service/repository_service.py

```diff
@@ -231,10 +232,14 @@ def inject_file_urls(node: AnyMutableJSON, *path: str) -> None:
                             inject_file_urls(child, *path)
                     else:
                         if url is None:
+                            pass
+                        elif url is url_injection_marker:
                             node['url'] = file_url_func(catalog=catalog,
                                                         fetch=False,
                                                         file_uuid=uuid,
                                                         version=version)
+                        else:
+                            assert False, url
```

Comment:
I'm happy to leave this in (the original intent was for this to be on-demand), but this change in semantics should be committed separately.

Reply:
I'm no longer motivated to include it now that we've removed `url_injection_marker`